src/fixate/reporting/csv.py

```diff
@@ -115,45 +115,53 @@ class CSVWriter:
     def __init__(self):
         self.csv_queue = Queue()
         self.csv_writer = None
-        self.reporting = CsvReporting()
+
+        self.exception_in_test = False
+        self.failed = False
+        self.chk_cnt = 0
+        self.csv_path = ""
+        self.test_module = None
+        self.start_time = None
+        self.current_test = None
+        self.data = fixate.config.get_config_dict()
+        self.data.update(fixate.config.get_plugin_data("plg_csv"))
 
     def install(self):
         self.csv_writer = ExcThread(
             target=self._csv_write, args=(self.csv_queue,), name="csv-writer"
         )
         self.csv_writer.start()
 
+        pub.subscribe(self.test_start, "Test_Start")
+        pub.subscribe(self.test_comparison, "Check")
+        pub.subscribe(self.test_exception, "Test_Exception")
+        pub.subscribe(self.test_complete, "Test_Complete")
+        pub.subscribe(self.sequence_update, "Sequence_Update")
+        pub.subscribe(self.sequence_complete, "Sequence_Complete")
+        pub.subscribe(self.user_wait_start, "UI_block_start")
+        pub.subscribe(self.user_wait_end, "UI_block_end")
+        pub.subscribe(self.driver_open, "driver_open")
+
     def uninstall(self):
+        pub.unsubscribe(self.test_start, "Test_Start")
+        pub.unsubscribe(self.test_comparison, "Check")
+        pub.unsubscribe(self.test_exception, "Test_Exception")
+        pub.unsubscribe(self.test_complete, "Test_Complete")
+        pub.unsubscribe(self.sequence_update, "Sequence_Update")
+        pub.unsubscribe(self.sequence_complete, "Sequence_Complete")
+        pub.unsubscribe(self.user_wait_start, "UI_block_start")
+        pub.unsubscribe(self.user_wait_end, "UI_block_end")
+
         if self.csv_writer:
             self.csv_queue.put(None)
             self.csv_writer.join()
         self.csv_writer = None
 
-    def _csv_write(self, cmd_q):
-        while True:
-            line = cmd_q.get()
-            if line is None:
-                break  # Command send to close csv_writer
-            try:
-                os.makedirs(os.path.dirname(self.reporting.csv_path))
-            except OSError as e:
-                pass
-            with open(self.reporting.csv_path, "a+", newline="") as f:
-                writer = csv.writer(f, quoting=csv.QUOTE_MINIMAL)
-                writer.writerow(line)
-
+    def is_alive(self):
```

Comment:
As in something like this? 
```
                try:
                    writer.writerow(line)
                except Exception as e:
                    self.exception = e
```

Reply:
Yes, but I'd raise that exception (possibly wrapped in a new exception)
```python
def is_alive():
    if self.csv_write is None:
        if self.exception is not None:
            raise self.exception
        else:
            return False
   else:
            return True
```

Actually I think `is_alive()` currently has a bug, because it will implicitly return `None` which will logically test as `False` if the thread is running?

Reply:
I've changed things up a bit to have is_alive raises an exception if one is saved. or if the thread has died without raising an exception.

Reply:
You can do `raise RuntimeError("bad thing logging module") from self.exception` so you get the best of both worlds.

Reply:
Ah! nice.